File: Interview Preparation Kit/Warm-up Challenges/Repeated String.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the repeatedString function below.
def repeatedString(s, n):

    a_instring = 0
    a_count = 0

    if n % len(s) == 0:
        for i in s:
            if i == 'a':
                a_instring += 1
        a_count = int(n / len(s)) * a_instring
    else:
        for i in s:
            if i == 'a':
                a_instring += 1
        a_count = int(n / len(s)) * a_instring

        a_instring = 0
        for i in range(n % len(s)):
            if s[i] == 'a':
                a_instring += 1
        a_count += a_instring
    
    return a_count


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    n = int(input())

    result = repeatedString(s, n)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Interview Preparation Kit/Warm-up Challenges/Repeated String.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the repeatedString function below.
long repeatedString(string s, long n) {

    long int count_instring, count_total;
    count_instring = count_total = 0;

    if ((n % s.length()) == 0){
        for (int i = 0; i < s.length(); i++){
            if (s[i] == 'a'){
                count_instring += 1;
            }
        }
        count_total = (n / s.length()) * count_instring;
        
    }
    else{
        for (int i = 0; i < s.length(); i++){
            if (s[i] == 'a'){
                count_instring += 1;
            }
        }
        count_total = (n / s.length()) * count_instring;

        count_instring = 0;

        for (int i = 0; i < (n % s.length()); i++){
            if (s[i] == 'a'){
                count_total += 1;
            }
        }
    }
    return count_total;

}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    long n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    long result = repeatedString(s, n);

    fout << result << "\n";

    fout.close();

    return 0;
}
